packages/aws-rfdk/lib/core/scripts/bash/installLustreClient.sh

```diff
@@ -0,0 +1,264 @@
+      esac
+      sudo yum -y install "https://downloads.whamcloud.com/public/lustre/lustre-${lustre_version}/el7/client/RPMS/x86_64/kmod-lustre-client-${lustre_version}-1.el7.x86_64.rpm"
+      sudo yum -y install "https://downloads.whamcloud.com/public/lustre/lustre-${lustre_version}/el7/client/RPMS/x86_64/lustre-client-${lustre_version}-1.el7.x86_64.rpm"
+    ;;
+
+    # RHEL 7.7, 7.8, and 7.9
+    # ----------------------
+    7.7|7.8|7.9)
+      architecture=$(uname -m)
+      case $architecture in
+        x86_64)
+          distro_name="el"
+          case $kernel_version in
+            ${rhel_kernel_versions[7.7]}) replace_ver="7.7";;
+            ${rhel_kernel_versions[7.8]}) replace_ver="7.8";;
+            ${rhel_kernel_versions[7.9]});; # Do nothing
+            *) echo "ERROR: Unsupported kernel version $kernel_version for RHEL $rhel_version"; exit 1;;
+          esac
+        ;;
+        aarch64)
+          distro_name="centos"
+          case $kernel_version in
+            # RHEL 7.8 for ARM-based AWS-Graviton powered instances have the 8.1 kernel version (see AWS docs)
+            ${rhel_kernel_versions[8.1]}) replace_ver="7.8";;
+            ${rhel_kernel_versions[8.2]});; # Do nothing
+            *) echo "ERROR: Unsupported kernel version $kernel_version for RHEL $rhel_version"; exit 1;;
+          esac
+        ;;
+        *) echo "ERROR: Unrecognized architecture: $architecture"; exit 1;;
+      esac
+
+      curl https://fsx-lustre-client-repo-public-keys.s3.amazonaws.com/fsx-rpm-public-key.asc -o /tmp/fsx-rpm-public-key.asc
+      sudo rpm --import /tmp/fsx-rpm-public-key.asc
+      sudo curl "https://fsx-lustre-client-repo.s3.amazonaws.com/${distro_name}/7/fsx-lustre-client.repo" -o /etc/yum.repos.d/aws-fsx.repo
+      if [[ -n "${replace_ver+x}" ]]; then
+        sudo sed -i "s#7#${replace_ver}#" /etc/yum.repos.d/aws-fsx.repo
+      fi
+
+      sudo yum clean all
+      sudo yum install -y kmod-lustre-client lustre-client
+    ;;
+
+    # RHEL 8.2 or newer
+    # -----------------
+    8.2|8.[1-9]*([0-9]))
+      case $kernel_version in
+        ${rhel_kernel_versions[8.2]}) replace_ver="8.2";;
+        ${rhel_kernel_versions[8.3]}  replace_ver="8.3";;
+        *)
+          if is_kernel_version_lower "$kernel_version" "${rhel_kernel_versions[8.2]}"; then
+            echo "ERROR: Kernel version ($kernel_version) is lower than the minimum required version: ${rhel_kernel_versions[8.2]}"
+            exit 1
+          fi
+        ;;
+      esac
+
+      curl https://fsx-lustre-client-repo-public-keys.s3.amazonaws.com/fsx-rpm-public-key.asc -o /tmp/fsx-rpm-public-key.asc
+      sudo rpm --import /tmp/fsx-rpm-public-key.asc
+      sudo curl https://fsx-lustre-client-repo.s3.amazonaws.com/el/8/fsx-lustre-client.repo -o /etc/yum.repos.d/aws-fsx.repo
+      if [[ -n "${replace_ver+x}" ]]; then
+        sudo sed -i "s#8#${replace_ver}#" /etc/yum.repos.d/aws-fsx.repo
+      fi
```

Comment:
Can you add a comment on what is going on here?

Reply:
Sure, added a brief comment explaining what it does. For more context, the contents of `aws-fsx.repo` will look something like this:
```
[aws-fsx]
name=AWS FSx Packages - $basearch
baseurl=https://fsx-lustre-client-repo.s3.amazonaws.com/el/8/$basearch/
enabled=1
gpgcheck=1

[aws-fsx-src]
name=AWS FSx Source
baseurl=https://fsx-lustre-client-repo.s3.amazonaws.com/el/8/SRPMS/
enabled=1
gpgcheck=1
```

This changes the `baseurl` to point to a URL that has the right versions of the lustre client packages for the kernel we're on